### DISCUSSION TITLE
Can Convolutional Layers Deal With Batch Trafffic Data?

### DISCUSSION THREAD

**Discussion by ZLXJC** [pyg-team/pytorch_geometric#4282] [Q&A]
I use the PEMS-04 Dataset in traffic prediction mission,and the dataset is divided into train and test. The dimension of train is [Batch,Node_number,Time_step,Feature].So,my question is if the cnvolutional layers in PyG are able to accept this kind of data and to predic the future time_step.

**Comment by rusty1s** [comment]
In general, the format of `[batch_size, #nodes, #time, #features]` is not understandable by PyG. PyG relies on two-dimensional matrices as node features. As such, it is best to merge both the batch dimension and the time dimension into the node feature dimension (`[batch_size * #nodes * #time, #features]`). This way, model weights are shared across all nodes in all examples and timestamps. After GNN processing, you can reshape the output node embeddings to your desired size.
